From: Tim Waugh <twaugh@redhat.com>
Date: Tue, 9 Aug 2016 18:11:07 +0200
Subject: Fix jobs with multiple files or multiple formats

Bug-Redhat: https://bugzilla.redhat.com/show_bug.cgi?id=972242
Bug: https://github.com/apple/cups/issues/4348
Patch-Name: fixes-for-jobs-with-multiple-files-and-multiple-formats.patch
---
 backend/ipp.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/backend/ipp.c b/backend/ipp.c
index 3f3e186..a99079e 100644
--- a/backend/ipp.c
+++ b/backend/ipp.c
@@ -1804,7 +1804,10 @@ main(int  argc,				/* I - Number of command-line args */
 	ippAddBoolean(request, IPP_TAG_OPERATION, "last-document",
         	      (i + 1) >= num_files);
 
-	if (document_format)
+	if (num_files > 1)
+	  ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_MIMETYPE,
+		       "document-format", NULL, "application/octet-stream");
+	else if (document_format)
 	  ippAddString(request, IPP_TAG_OPERATION, IPP_TAG_MIMETYPE,
 		       "document-format", NULL, document_format);
 
